lib/src/model/http_log/http_log_storage.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:collection/collection.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:lichess_mobile/src/db/database.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:sqflite/sqflite.dart';
+
+part 'http_log_storage.g.dart';
+part 'http_log_storage.freezed.dart';
+
+/// Provides an instance of [HttpLogStorage] using Riverpod.
+@Riverpod(keepAlive: true)
+Future<HttpLogStorage> httpLogStorage(Ref ref) async {
+  final db = await ref.watch(databaseProvider.future);
+  return HttpLogStorage(db);
+}
+
+const kHttpLogStorageTable = 'http_log';
+
+/// Manages the storage of HTTP logs in a SQLite database.
+class HttpLogStorage {
+  const HttpLogStorage(this._db);
+  final Database _db;
+
+  /// Retrieves a paginated list of [HttpLog] entries from the database.
+  Future<HttpLogs> page({int? cursor, int limit = 100}) async {
+    final res = await _db.query(
+      kHttpLogStorageTable,
+      limit: limit + 1,
+      orderBy: 'id DESC',
+      where: cursor != null ? 'id <= $cursor' : null,
+    );
+    return HttpLogs(
+      items: res.take(limit).map(HttpLog.fromJson).toIList(),
+      next: res.elementAtOrNull(limit)?['id'] as int?,
+    );
+  }
+
+  /// Saves an [HttpLog] entry to the database.
+  Future<void> save(HttpLog httpLog) async {
+    await _db.insert(kHttpLogStorageTable, {
+      ...httpLog.toJson(),
+      'lastModified': DateTime.now().toIso8601String(),
+    }, conflictAlgorithm: ConflictAlgorithm.replace);
+  }
+
+  Future<void> update(
```

Comment:
I don't think we need this, do we?

Reply:
At the moment http logs are saved when a request is made and then updated with the status code when the response is received.

Reply:
No I just thought the log would never need to be updated.

Reply:
Yeah maybe was better to save the log directly when receiving the response...only difference is that we cannot save unanswered http requests.